main.go

```diff
@@ -106,34 +118,40 @@ func collectMetrics(scheme string, host string, apiKey string) *prometheus.Regis
 }
 
 func main() {
-	parseFlagsAndEnv()
 
+ parseFlagsAndEnv()
+	
 	http.HandleFunc("/metrics", func(response http.ResponseWriter, request *http.Request) {
+	
+
 		host := request.URL.Query().Get("host")
 		apiKey := request.URL.Query().Get("apiKey")
-		scheme := request.URL.Query().Get("scheme")
-
+		// why is the scheme formed from the request here? 
+		//scheme := request.URL.Query().Get("scheme")
```

Comment:
There are 3 ways of passing data to the exporter:

1. By using query parameter (e.g. calling it via `/exporter?host=mail.foobar.com`). This allows users to configure the exporter in their prometheus config instead of having separate configuration for the exporter.
2. Via command line arguments
3. Via environment variables

They are used in that order: Query Parameters override CLI Arguments override Environment

Reply:
I see, the problem is that this empties scheme after it is set either via the variable or on the command line. 

Maybe do something like 

                if scheme == "" { 
                        scheme := request.URL.Query().Get("scheme")
                }

Reply:
Ah, I see the problem: If you check the other parameters (e.g. `host`), the flow is as follows:

* The environment variable is read into `envHost`, which is used as a default value for the `--host` command line argument
* The command line argument is read into `defaultHost`, which is used as a default value for `host`